test_suite.py

```diff
@@ -21,12 +21,32 @@
 # Used for file "streaming" only.
 REALTIME_RESOLUTION = 0.250
 
+subtitle_line_counter = 0
+
+def subtitle_time_formatter(seconds, separator):
+    hours = int(seconds // 3600)
+    minutes = int((seconds % 3600) // 60)
+    secs = int(seconds % 60)
+    millis = int((seconds - int(seconds)) * 1000)
+    return f"{hours:02}:{minutes:02}:{secs:02}{separator}{millis:03}"
+
+def subtitle_formatter(response, separator="."):
+    global subtitle_line_counter
+    start = response['start']
+    end = start + response['duration']
+    transcript = response.get('channel', {})\
+        .get('alternatives', [{}])[0]\
+        .get('transcript', '')
+    subtitle_line_counter += 1
+    subtitle_string = f"{subtitle_line_counter}\n{subtitle_time_formatter(start, separator)} --> {subtitle_time_formatter(end, separator)}\n- {transcript}\n\n"
```

Comment:
According to the SRT examples I'm seeing in the links you very helpfully provided, there should not be a hyphen at the beginning of the third line. I think the differences would be easier to spot if you split up this line.

```suggestion
    subtitle_string = f"{subtitle_line_counter}\n"
    subtitle_string += f"{subtitle_time_formatter(start, separator)} --> "
    subtitle_string += f"{subtitle_time_formatter(end, separator)}\n"
    if format == 'vtt':
        subtitle_string += "- "
    subtitle_string += f"{transcript}\n\n"
```

I'd suggest passing the `format` as an argument to this function, since you already have it in the caller as `output`.

Reply:
Good catch! Split up the line as you suggested. I passed in `format` and then had it set `separator` within the function based on the `format`.